Bug: panic when call gws connection close

### Describe the bug

when using gws connection, and calling close(), it paniced
```
[invalid memory address or nil pointer dereference](panic: runtime error: invalid memory address or nil pointer dereference)
```

### Reproduction

call db.close(ctx)

### Steps to reproduce

call db.close(ctx) after connect

### Expected behaviour

not panic

### Error

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x1035725e4]

goroutine 1 [running]:
github.com/surrealdb/surrealdb.go/pkg/connection/gws.(*Connection).Close(0x140001e7a40, {0x2d0b2fb?, 0x10457edc0?})
	/Users/carterwang/go/pkg/mod/github.com/surrealdb/surrealdb.go@v0.6.1-0.20250801123350-310e9993ecce/pkg/connection/gws/websocket.go:194 +0x224
github.com/surrealdb/surrealdb%2ego.(*DB).Close(...)
	/Users/carterwang/go/pkg/mod/github.com/surrealdb/surrealdb.go@v0.6.1-0.20250801123350-310e9993ecce/db.go:138
```

### SDK version

v0.6.1-0.20250801123350-310e9993ecce

### SurrealDB version

v2.3.7

### Contact Details

_No response_

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/example/connection.go
+++ b/example/connection.go
@@ -11,7 +11,7 @@ import (
 )
 
 const (
-	defaultURL = "ws://localhost:8000"
+	defaultWSURL = "ws://localhost:8000"
 )
 
 var (
@@ -20,9 +20,9 @@ var (
 	useGWS     = os.Getenv("SURREALDB_CONNECTION_IMPL") == "gws"
 )
 
-func getSurrealDBWSURL() string {
+func getSurrealDBURL() string {
 	if currentURL == "" {
-		return defaultURL
+		return defaultWSURL
 	}
 	return currentURL
 }
@@ -32,6 +32,13 @@ func getSurrealDBHTTPURL() string {
 		return "http://localhost:8000"
 	}
 	return strings.ReplaceAll(currentURL, "ws", "http")
+}
+
+func getSurrealDBWSURL() string {
+	if currentURL == "" {
+		return defaultWSURL
+	}
+	return strings.ReplaceAll(currentURL, "http", "ws")
 }
 
 func newSurrealDBWSConnection(database string, tables ...string) *surrealdb.DB {
@@ -47,7 +54,7 @@ func newSurrealDBWSConnection(database string, tables ...string) *surrealdb.DB {
 	var db *surrealdb.DB
 
 	if useGWS {
-		p, confErr := surrealdb.Configure(getSurrealDBWSURL(),
+		p, confErr := surrealdb.Configure(getSurrealDBURL(),
 			surrealdb.WithReconnectionCheckInterval(reconnectDuration),
 		)
 		if confErr != nil {
@@ -62,7 +69,7 @@ func newSurrealDBWSConnection(database string, tables ...string) *surrealdb.DB {
 		var err error
 		db, err = surrealdb.Connect(
 			context.Background(),
-			getSurrealDBWSURL(),
+			getSurrealDBURL(),
 			surrealdb.WithReconnectionCheckInterval(reconnectDuration),
 		)
 		if err != nil {

--- a/example/example_db_signin_test.go
+++ b/example/example_db_signin_test.go
@@ -10,7 +10,7 @@ import (
 func ExampleDB_signin_failure() {
 	db, err := surrealdb.Connect(
 		context.Background(),
-		getSurrealDBWSURL(),
+		getSurrealDBURL(),
 	)
 	if err != nil {
 		panic(err)

--- a/example/example_gws_connection_test.go
+++ b/example/example_gws_connection_test.go
@@ -5,16 +5,24 @@ import (
 	"fmt"
 
 	surrealdb "github.com/surrealdb/surrealdb.go"
+	"github.com/surrealdb/surrealdb.go/pkg/connection/gws"
 )
 
-func ExampleDB_signin_failure() {
-	db, err := surrealdb.Connect(
-		context.Background(),
+func ExampleConnection_gws() {
+	conf, err := surrealdb.Configure(
 		getSurrealDBWSURL(),
 	)
+	conf.Logger = nil // Disable logging for this example
 	if err != nil {
 		panic(err)
 	}
+
+	conn := gws.New(conf)
+	if connErr := conn.Connect(context.Background()); connErr != nil {
+		panic(fmt.Sprintf("Failed to connect: %v", connErr))
+	}
+
+	db := surrealdb.New(conn)
 
 	// Attempt to sign in without setting namespace or database
 	// This should fail with an error, whose message will depend on the connection type.

--- a/pkg/connection/gws/websocket.go
+++ b/pkg/connection/gws/websocket.go
@@ -116,7 +116,7 @@ func (c *Connection) handleResponse(data []byte) {
 		// Instead, we just log the error and return,
 		// in the hope that the caller notices the programming error
 		// by the RPC timing out, and the error being logged here.
-		c.Logger.Error(
+		c.logError(
 			fmt.Sprintf("error in response: %v", rpcRes.Error),
 			"result", fmt.Sprint(rpcRes.Result),
 		)
@@ -156,6 +156,7 @@ func New(params *connection.Config) *Connection {
 	conn.Unmarshaler = params.Unmarshaler
 	conn.ResponseChannels = make(map[string]chan connection.RPCResponse[cbor.RawMessage])
 	conn.NotificationChannels = make(map[string]chan connection.Notification)
+	conn.Logger = params.Logger
 
 	return conn
 }
@@ -187,11 +188,11 @@ func (c *Connection) Close(ctx context.Context) error {
 
 	err := c.conn.WriteClose(constants.CloseMessageCode, []byte(""))
 	if err != nil {
-		c.Logger.Error("failed to close WebSocket connection", "error", err)
+		c.logError("failed to close WebSocket connection", "error", err)
 	}
 
 	if err := c.conn.NetConn().Close(); err != nil {
-		c.Logger.Error("failed to close underlying network connection", "error", err)
+		c.logError("failed to close underlying network connection", "error", err)
 	}
 	c.conn = nil
 
@@ -239,7 +240,7 @@ func (c *Connection) Connect(ctx context.Context) error {
 	conn, resp, err := gws.NewClient(c.handler, option)
 	if resp != nil {
 		if closeErr := resp.Body.Close(); closeErr != nil {
-			c.Logger.Error("failed to close response body", "error", closeErr)
+			c.logError("failed to close response body", "error", closeErr)
 		}
 	}
 	if err != nil {
@@ -345,4 +346,10 @@ func (c *Connection) Unset(ctx context.Context, key string) error {
 // Use implements connection.Connection.
 func (c *Connection) Use(ctx context.Context, namespace, database string) error {
 	return connection.Send[any](c, ctx, nil, "use", namespace, database)
+}
+
+func (c *Connection) logError(msg string, args ...any) {
+	if c.Logger != nil {
+		c.Logger.Error(msg, args...)
+	}
 }

